packages/core/src/external/carequality/pd/validating-pd.ts

```diff
@@ -0,0 +1,189 @@
+import {
+  Address as FHIRAddress,
+  ContactPoint,
+  Identifier,
+  Patient as FHIRPatient,
+} from "@medplum/fhirtypes";
+import { getStateEnum, USState } from "../../../domain/geographic-locations";
+import { Address } from "../../../domain/address";
+import { Contact } from "../../../domain/contact";
+import { PatientData, PersonalIdentifier } from "../../../domain/patient";
+import { isContactType } from "../../fhir/patient/index";
+import { MetriportError } from "../../../util/error/metriport-error";
+import status from "http-status";
+
+export class IHEGatewayError extends MetriportError {
+  constructor(
+    message: string,
+    public iheErrorCode: string,
+    statusCode: number = status.BAD_REQUEST
+  ) {
+    super(message);
+    this.name = this.constructor.name;
+    this.status = statusCode;
+  }
+}
+
+export class PatientAddressRequestedError extends IHEGatewayError {
+  constructor(message = "Address Line 1 is not defined") {
```

Comment:
ditto: missing `cause`

Reply:
Just to confirm, don't we want this to represent a client error? If so, we should set the status to 400 right?

Reply:
updated the PatientAddressRequestedError to have status 400